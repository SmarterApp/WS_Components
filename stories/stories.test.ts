import initStoryshots from "@storybook/addon-storyshots";

// Note: Storyshots doesnt not use JsDom causing some tests to fail.
// The following regex will run all storyShot tests except for the ones specified.
initStoryshots({
  storyKindRegex: /^((?!.*?(Item Table Container|Filter Link|Search Result Container)).)*$/
});
