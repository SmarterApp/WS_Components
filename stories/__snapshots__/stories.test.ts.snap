// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots AdvancedFilter multiple filter items 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexFlow": "column nowrap",
      "height": "100%",
      "justifyContent": "center",
      "paddingTop": "300px",
      "width": "100%",
    }
  }
>
  <div
    className="filter-selection"
    id="grade 3-filter"
  >
    <label>
      <span
        info-label={true}
      >
        Grade 3
      </span>
      <span
        data-tooltip="Grade 3"
        data-tooltip-position="top"
      >
        <span
          className="fa fa-info-circle fa-sm"
        />
      </span>
    </label>
    <div
      className="child-filter-container"
    >
      <div
        className="filter-all-btn-container"
      >
        <button
          className="filter-all-btn  selected-button filter-button"
          onClick={[Function]}
        >
          All
        </button>
      </div>
      <div
        className="filter-button-container"
      >
        <button
          className=" filter-button"
          onClick={[Function]}
        >
          Grade 1
        </button>
        <button
          className=" filter-button"
          onClick={[Function]}
        >
          Grade 2
        </button>
        <button
          className=" filter-button"
          onClick={[Function]}
        >
          Grade 3
        </button>
        <button
          className=" filter-button"
          onClick={[Function]}
        >
          Grade 4
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots AdvancedFilter one filter item 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexFlow": "column nowrap",
      "height": "100%",
      "justifyContent": "center",
      "paddingTop": "300px",
      "width": "100%",
    }
  }
>
  <div
    className="filter-selection"
    id="grade 3-filter"
  >
    <label>
      <span
        info-label={true}
      >
        Grade 3
      </span>
      <span
        data-tooltip="Grade 3"
        data-tooltip-position="top"
      >
        <span
          className="fa fa-info-circle fa-sm"
        />
      </span>
    </label>
    <div
      className="child-filter-container"
    >
      <div
        className="filter-all-btn-container"
      >
        <button
          className="filter-all-btn  selected-button filter-button"
          onClick={[Function]}
        >
          All
        </button>
      </div>
      <div
        className="filter-button-container"
      >
        <button
          className=" filter-button"
          onClick={[Function]}
        >
          Grade 3
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots AdvancedFilterContainer normal render 1`] = `
<div
  className="advanced-filter-container"
>
  <div
    className="filter-sub-header-container"
  >
    <div
      className="filter-advanced-filter-header"
    >
      <div
        className="filter-advanced-filter-title"
      >
        <h2
          style={
            Object {
              "color": "#63666A",
            }
          }
        >
          <span
            className="fa fa-tasks fa-lg"
          />
           Advanced Filters
        </h2>
        <span
          style={
            Object {
              "marginTop": "6px",
            }
          }
        >
           Click on an item to remove it from the list
        </span>
      </div>
      <div
        style={
          Object {
            "display": "flex",
            "marginRight": "10px",
          }
        }
      >
        <button
          className="filter-button"
          onClick={[Function]}
        >
          Expand 
          <span
            className="fa fa-chevron-right"
          />
        </button>
      </div>
    </div>
    <div
      className="filter-header"
    >
      <div
        className="filter-status"
      />
    </div>
  </div>
</div>
`;

exports[`Storyshots BasicFilter DropDown: multiple filter items 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexFlow": "column nowrap",
      "height": "100%",
      "justifyContent": "center",
      "paddingTop": "300px",
      "width": "100%",
    }
  }
>
  <div
    className="basic-filter-selection"
    id="grade 3-basic-filter"
  >
    <label>
      Grade 3
    </label>
    <div
      className="basic-filter-body"
    >
      <select
        onChange={[Function]}
      >
        <option
          value="12346"
        >
          Grade 1
        </option>
        <option
          value="12347"
        >
          Grade 2
        </option>
        <option
          value="12345"
        >
          Grade 3
        </option>
        <option
          value="12348"
        >
          Grade 4
        </option>
      </select>
    </div>
  </div>
</div>
`;

exports[`Storyshots BasicFilter DropDown: one filter item 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexFlow": "column nowrap",
      "height": "100%",
      "justifyContent": "center",
      "paddingTop": "300px",
      "width": "100%",
    }
  }
>
  <div
    className="basic-filter-selection"
    id="grade 3-basic-filter"
  >
    <label>
      Grade 3
    </label>
    <div
      className="basic-filter-body"
    >
      <select
        onChange={[Function]}
      >
        <option
          value="12345"
        >
          Grade 3
        </option>
      </select>
    </div>
  </div>
</div>
`;

exports[`Storyshots BasicFilter RadioBtn: multiple filter items 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexFlow": "column nowrap",
      "height": "100%",
      "justifyContent": "center",
      "paddingTop": "300px",
      "width": "100%",
    }
  }
>
  <div
    className="basic-filter-selection"
    id="grade 3-basic-filter"
  >
    <label>
      Grade 3
    </label>
    <div
      className="basic-filter-body"
    >
      <form>
        <label>
          <input
            checked={false}
            onChange={[Function]}
            type="radio"
            value="12346"
          />
          Grade 1
        </label>
        <label>
          <input
            checked={false}
            onChange={[Function]}
            type="radio"
            value="12347"
          />
          Grade 2
        </label>
        <label>
          <input
            checked={false}
            onChange={[Function]}
            type="radio"
            value="12345"
          />
          Grade 3
        </label>
        <label>
          <input
            checked={false}
            onChange={[Function]}
            type="radio"
            value="12348"
          />
          Grade 4
        </label>
      </form>
    </div>
  </div>
</div>
`;

exports[`Storyshots BasicFilter RadioBtn: one filter item 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexFlow": "column nowrap",
      "height": "100%",
      "justifyContent": "center",
      "paddingTop": "300px",
      "width": "100%",
    }
  }
>
  <div
    className="basic-filter-selection"
    id="grade 3-basic-filter"
  >
    <label>
      Grade 3
    </label>
    <div
      className="basic-filter-body"
    >
      <form>
        <label>
          <input
            checked={false}
            onChange={[Function]}
            type="radio"
            value="12345"
          />
          Grade 3
        </label>
      </form>
    </div>
  </div>
</div>
`;

exports[`Storyshots BasicFilterContainer Basic Filter only 1`] = `
<div
  className="basic-filter-container"
>
  <div
    className="basic-filter"
  >
    <div
      className="basic-filter-selection"
      id="grade-basic-filter"
    >
      <label>
        Grade
      </label>
      <div
        className="basic-filter-body"
      >
        <select
          onChange={[Function]}
        >
          <option
            value="7"
          >
            Grades 3-5
          </option>
          <option
            value="56"
          >
            Grades 6-8
          </option>
          <option
            value="960"
          >
            High School
          </option>
        </select>
      </div>
    </div>
    <div
      className="basic-filter-selection"
      id="subjects-basic-filter"
    >
      <label>
        Subjects
      </label>
      <div
        className="basic-filter-body"
      >
        <form>
          <label>
            <input
              checked={false}
              onChange={[Function]}
              type="radio"
              value="ELA"
            />
            English Literacy/ Arts
          </label>
          <label>
            <input
              checked={false}
              onChange={[Function]}
              type="radio"
              value="Math"
            />
            Mathematics
          </label>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots BasicFilterContainer Basic Filter with Advanced filter expansion button 1`] = `
<div
  className="basic-filter-container"
>
  <div
    className="basic-filter"
  >
    <div
      className="basic-filter-selection"
      id="grade-basic-filter"
    >
      <label>
        Grade
      </label>
      <div
        className="basic-filter-body"
      >
        <select
          onChange={[Function]}
        >
          <option
            value="7"
          >
            Grades 3-5
          </option>
          <option
            value="56"
          >
            Grades 6-8
          </option>
          <option
            value="960"
          >
            High School
          </option>
        </select>
      </div>
    </div>
    <div
      className="basic-filter-selection"
      id="subjects-basic-filter"
    >
      <label>
        Subjects
      </label>
      <div
        className="basic-filter-body"
      >
        <form>
          <label>
            <input
              checked={false}
              onChange={[Function]}
              type="radio"
              value="ELA"
            />
            English Literacy/ Arts
          </label>
          <label>
            <input
              checked={false}
              onChange={[Function]}
              type="radio"
              value="Math"
            />
            Mathematics
          </label>
        </form>
      </div>
    </div>
  </div>
  <div
    className="basic-filter-button-container"
  >
    <div>
      Advanced Filters
    </div>
    <button
      className="filter-button"
      onClick={[Function]}
    >
      Show
       
      <span
        className="fa fa-chevron-right"
      />
    </button>
  </div>
</div>
`;

exports[`Storyshots Filter Container Contains both Basic and Advanced Filter 1`] = `
<div
  style={
    Object {
      "width": "100%",
    }
  }
>
  <div
    className="basic-filter-container"
  >
    <div
      className="basic-filter"
    >
      <div
        className="basic-filter-selection"
        id="grade-basic-filter"
      >
        <label>
          Grade
        </label>
        <div
          className="basic-filter-body"
        >
          <select
            onChange={[Function]}
          >
            <option
              value="7"
            >
              Grades 3-5
            </option>
            <option
              value="56"
            >
              Grades 6-8
            </option>
            <option
              value="960"
            >
              High School
            </option>
          </select>
        </div>
      </div>
      <div
        className="basic-filter-selection"
        id="subjects-basic-filter"
      >
        <label>
          Subjects
        </label>
        <div
          className="basic-filter-body"
        >
          <form>
            <label>
              <input
                checked={false}
                onChange={[Function]}
                type="radio"
                value="ELA"
              />
              English Literacy/ Arts
            </label>
            <label>
              <input
                checked={false}
                onChange={[Function]}
                type="radio"
                value="Math"
              />
              Mathematics
            </label>
          </form>
        </div>
      </div>
    </div>
    <div
      className="basic-filter-button-container"
    >
      <div>
        Advanced Filters
      </div>
      <button
        className="filter-button"
        onClick={[Function]}
      >
        Show
         
        <span
          className="fa fa-chevron-right"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`Storyshots Layout name links 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexFlow": "column nowrap",
      "height": "100%",
      "justifyContent": "center",
      "paddingTop": "300px",
      "width": "100%",
    }
  }
>
  <div
    className="site-body"
  >
    <header
      role="navigation"
    >
      <div
        className="skip-link"
        id="skip-main"
      >
        <a
          className="skip-link"
          href="/#main"
          onClick={[Function]}
        >
          Skip to main content
        </a>
      </div>
      <nav
        role="navigation"
      >
        <div
          className="sbNav"
        >
          <div
            className="sbNav-titleGroup"
          >
            <div
              className="sbNav-titleGroup-item"
            >
              <a
                href="http://www.smarterbalanced.org/"
                target="_blank"
                title="Smarter Balanced Home"
              >
                <img
                  alt="Smarter Balanced Logo"
                  src="/Images/SmarterBalanced_logo.png"
                />
              </a>
            </div>
            <div
              className="sbNav-titleGroup-item"
            >
              <h3>
                Test
              </h3>
            </div>
          </div>
          <div
            className="sbNav-linksGroup"
          >
            <a
              className="sbNav-linksGroup-item"
              href="url1"
              onClick={[Function]}
              style={undefined}
            >
              <span
                className=""
              />
               
              link1
            </a>
            <a
              className="sbNav-linksGroup-item"
              href="url2"
              onClick={[Function]}
              style={undefined}
            >
              <span
                className=""
              />
               
              link2
            </a>
          </div>
        </div>
      </nav>
    </header>
    <main
      className="container site-content"
      id="main"
      role="main"
    />
    <footer
      role="contentinfo"
    >
      <a
        className="footer-sbLink footer-practiceTests"
        href="http://practice.smarterbalanced.org/"
      >
        Practice and training tests are available here.
      </a>
      <div
        className="footer-copyright"
      >
        <p>
          © The Regents of the University of California – Smarter Balanced Assessment Consortium 
        </p>
        <a
          className="footer-sbLink"
          href="//www.smarterbalanced.org"
        >
           www.SmarterBalanced.org
        </a>
      </div>
    </footer>
  </div>
</div>
`;

exports[`Storyshots Layout name no links 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexFlow": "column nowrap",
      "height": "100%",
      "justifyContent": "center",
      "paddingTop": "300px",
      "width": "100%",
    }
  }
>
  <div
    className="site-body"
  >
    <header
      role="navigation"
    >
      <div
        className="skip-link"
        id="skip-main"
      >
        <a
          className="skip-link"
          href="/#main"
          onClick={[Function]}
        >
          Skip to main content
        </a>
      </div>
      <nav
        role="navigation"
      >
        <div
          className="sbNav"
        >
          <div
            className="sbNav-titleGroup"
          >
            <div
              className="sbNav-titleGroup-item"
            >
              <a
                href="http://www.smarterbalanced.org/"
                target="_blank"
                title="Smarter Balanced Home"
              >
                <img
                  alt="Smarter Balanced Logo"
                  src="/Images/SmarterBalanced_logo.png"
                />
              </a>
            </div>
            <div
              className="sbNav-titleGroup-item"
            >
              <h3>
                Test
              </h3>
            </div>
          </div>
        </div>
      </nav>
    </header>
    <main
      className="container site-content"
      id="main"
      role="main"
    />
    <footer
      role="contentinfo"
    >
      <a
        className="footer-sbLink footer-practiceTests"
        href="http://practice.smarterbalanced.org/"
      >
        Practice and training tests are available here.
      </a>
      <div
        className="footer-copyright"
      >
        <p>
          © The Regents of the University of California – Smarter Balanced Assessment Consortium 
        </p>
        <a
          className="footer-sbLink"
          href="//www.smarterbalanced.org"
        >
           www.SmarterBalanced.org
        </a>
      </div>
    </footer>
  </div>
</div>
`;

exports[`Storyshots LayoutNav name links 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexFlow": "column nowrap",
      "height": "100%",
      "justifyContent": "center",
      "paddingTop": "300px",
      "width": "100%",
    }
  }
>
  <header
    role="navigation"
  >
    <div
      className="skip-link"
      id="skip-main"
    >
      <a
        className="skip-link"
        href="/#main"
        onClick={[Function]}
      >
        Skip to main content
      </a>
    </div>
    <nav
      role="navigation"
    >
      <div
        className="sbNav"
      >
        <div
          className="sbNav-titleGroup"
        >
          <div
            className="sbNav-titleGroup-item"
          >
            <a
              href="http://www.smarterbalanced.org/"
              target="_blank"
              title="Smarter Balanced Home"
            >
              <img
                alt="Smarter Balanced Logo"
                src="/Images/SmarterBalanced_logo.png"
              />
            </a>
          </div>
          <div
            className="sbNav-titleGroup-item"
          >
            <h3>
              Test
            </h3>
          </div>
        </div>
        <div
          className="sbNav-linksGroup"
        >
          <a
            className="sbNav-linksGroup-item"
            href="url1"
            onClick={[Function]}
            style={undefined}
          >
            <span
              className=""
            />
             
            link1
          </a>
          <a
            className="sbNav-linksGroup-item"
            href="url2"
            onClick={[Function]}
            style={undefined}
          >
            <span
              className=""
            />
             
            link2
          </a>
        </div>
      </div>
    </nav>
  </header>
</div>
`;

exports[`Storyshots LayoutNav name no links 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexFlow": "column nowrap",
      "height": "100%",
      "justifyContent": "center",
      "paddingTop": "300px",
      "width": "100%",
    }
  }
>
  <header
    role="navigation"
  >
    <div
      className="skip-link"
      id="skip-main"
    >
      <a
        className="skip-link"
        href="/#main"
        onClick={[Function]}
      >
        Skip to main content
      </a>
    </div>
    <nav
      role="navigation"
    >
      <div
        className="sbNav"
      >
        <div
          className="sbNav-titleGroup"
        >
          <div
            className="sbNav-titleGroup-item"
          >
            <a
              href="http://www.smarterbalanced.org/"
              target="_blank"
              title="Smarter Balanced Home"
            >
              <img
                alt="Smarter Balanced Logo"
                src="/Images/SmarterBalanced_logo.png"
              />
            </a>
          </div>
          <div
            className="sbNav-titleGroup-item"
          >
            <h3>
              Test
            </h3>
          </div>
        </div>
      </div>
    </nav>
  </header>
</div>
`;
